File: 4005-maximum-total-subarray-value-i/maximum-total-subarray-value-i.cpp
class Solution {
public:
    long long maxTotalValue(vector<int>& nums, int k) {
        if(nums.empty()) return 0;
        
        // Find both the minimum and maximum elements in a single pass
        auto [min_it, max_it] = minmax_element(nums.begin(), nums.end());

        // Cast to long long to prevent potential integer overflow during multiplication
        return static_cast<long long>(*max_it - *min_it) * k;
    }
};

File: 4005-maximum-total-subarray-value-i/maximum-total-subarray-value-i.js
var maxTotalValue = function (nums, k) {
    let m1 = Infinity, m2 = -Infinity
    for (const x of nums) {
        m1 = Math.min(m1, x)
        m2 = Math.max(m2, x)
    }
    return (m2 - m1) * k
};